rosetta/retriever/retriever_test.go

```diff
@@ -869,7 +873,114 @@ func TestRetriever_Transaction(t *testing.T) {
 
 		ret := retriever.BaselineRetriever(t, retriever.WithConverter(convert))
 
-		_, err := ret.Transaction(rosBlockID, mocks.GenericTransactionQualifier(0))
+		_, err := ret.Transaction(mocks.GenericRosBlockID, mocks.GenericTransactionQualifier(0))
+
+		assert.Error(t, err)
+	})
+}
+
+func TestRetriever_Sequence(t *testing.T) {
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		validator := mocks.BaselineValidator(t)
+		validator.BlockFunc = func(blockID identifier.Block) (uint64, flow.Identifier, error) {
+			assert.Equal(t, mocks.GenericRosBlockID, blockID)
+
+			return mocks.GenericHeight, flow.ZeroID, nil
+		}
+		validator.AccountFunc = func(accountID identifier.Account) (flow.Address, error) {
+			assert.Equal(t, mocks.GenericAccountID(0), accountID)
+
+			return mocks.GenericAddress(0), nil
+		}
+
+		invoker := mocks.BaselineInvoker(t)
+		invoker.KeyFunc = func(height uint64, address flow.Address, index int) (*flow.AccountPublicKey, error) {
+			assert.Equal(t, mocks.GenericHeight, height)
+			assert.Equal(t, mocks.GenericAddress(0), address)
+			assert.Equal(t, 0, index)
+
+			return &mocks.GenericAccount.Keys[0], nil
+		}
+
+		ret := retriever.New(
+			mocks.GenericParams,
+			mocks.BaselineReader(t),
+			validator,
+			mocks.BaselineGenerator(t),
+			invoker,
+			mocks.BaselineConverter(t),
+		)
+
+		seqNum, err := ret.Sequence(mocks.GenericRosBlockID, mocks.GenericAccountID(0), 0)
```

Comment:
So are we back to not declaring things at the top of the function/section?

Wouldn't it make more sense to have `key := mocks.GenericAccount.Keys[0]`? And the rest declared at the top?